### PR TITLE
Split protos into data.proto and events.proto

### PR DESCRIPTION
**Pull request by jskeet**
cc @yolocs 

I acknowledge we may want to work on the README a bit further - potentially removing the term "envelope" at some point. (I think it's a useful term, but it's not part of CloudEvents terminology. The tricky bit is referring to "CloudEvent message" without being ambiguous I suspect.) However, I'd really like to get this merged ASAP to unblock client library generation.

### CODE REVIEW

**Comment by yolocs** [comment]
Thanks, Jon! 
/lgtm

**Comment by jskeet** [comment]
Will wait for Grant to review - then I'll need to update the .NET repo accordingly.

File: proto/README.md [hunk 1]
```diff
@@ -24,6 +25,11 @@ something happening to a Cloud Storage object uses `StorageObjectData`
 as the data message. Each envelope message specifies the CloudEvent
 type it's associated with via a proto annotation.
 
+Initially we will generate libraries that only contain the data
+messages. The event messages serve as useful documentation for the
+event type strings and extension attributes. They will be included
+in the libraries at a later date if that proves useful.
```
Comment (grant) [inline proto/README.md:31]: I generally don't include "we will" statements in READMEs, and rather add TODOs and link to GitHub issues.

But this is fine.

File: proto/google/events/cloud/scheduler/v1/data.proto [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+package google.events.cloud.scheduler.v1;
+
+option csharp_namespace = "Google.Events.Protobuf.Cloud.Scheduler.V1";
+
+// Scheduler event data.
+message SchedulerData {
+  // The custom data the user specified when creating the scheduler source.
+  bytes custom_data = 1;
+}
```
Comment (grant) [inline proto/google/events/cloud/scheduler/v1/data.proto:25]: nit: looks like we have newline inconsistencies? No newline here.

File: proto/google/events/cloud/storage/v1/data.proto [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  // [https://tools.ietf.org/html/rfc7231#section-3.1.2.2][RFC 7231 §3.1.2.2]
+  string content_encoding = 1;
+
+  // Content-Disposition of the object data, matching
+  // [https://tools.ietf.org/html/rfc6266][RFC 6266].
+  string content_disposition = 2;
+
+  // Cache-Control directive for the object data, matching
+  // [https://tools.ietf.org/html/rfc7234#section-5.2"][RFC 7234 §5.2].
+  string cache_control = 3;
```
Comment (grant) [inline proto/google/events/cloud/storage/v1/data.proto:45]: Is this file copied? Is there a reason to skip 4?

Comment (jskeet) [reply proto/google/events/cloud/storage/v1/data.proto:45]: Yes, it's copied (with some changes) from https://github.com/googleapis/googleapis/blob/master/google/storage/v1/storage_resources.proto

Where changes aren't deliberate and required, I'd prefer to stick to what's there, making it easier to diff.

File: proto/google/events/cloud/scheduler/v1/data.proto (hunk 2 above)
Comment (jskeet) [reply proto/google/events/cloud/scheduler/v1/data.proto:25]: Not deliberately. Fixed.

**Comment by jskeet** [comment]
All files are "hand-written to an extent", but most have started from another source. If we want to document that somewhere, let's do it in a separate PR. Feel free to file an issue and assign it to me.

**Pull request merged**